### ISSUE TITLE
[Feature]: It should be possible to see video preview for video attachment with custom name

### ISSUE THREAD

**Issue by RemigiuszDudek** [microsoft/playwright#33073]
### 🚀 Feature Request

Giving a custom name to the attachment should not prevent displaying a video preview on the report.

Currently one needs to name all video attachments as `video`, only then preview is available:
```await testInfo.attach('video', { path, contentType: 'video/webm' });```
![Image](https://github.com/user-attachments/assets/e14f02f9-c2f6-436c-bb08-bdb1ed0d6de6)

If one gives more meaningful names, preview is not available:
```await testInfo.attach('customName', { path, contentType: 'video/webm' });```
![Image](https://github.com/user-attachments/assets/19b36e6c-3194-4634-b997-921807eeabb7)

I found that `html-reporter` in [testResultViewer.tsx](https://github.com/microsoft/playwright/blob/main/packages/html-reporter/src/testResultView.tsx) recognizes if a given attachment is a video attachment through name:
```
const screenshots = new Set(attachments.filter(a => a.contentType.startsWith('image/')));
const videos = attachments.filter(a => a.name === 'video'); // <---- here is the code I refer to
const traces = attachments.filter(a => a.name === 'trace');
const htmls = attachments.filter(a => a.contentType.startsWith('text/html'));
```

The suggestion is to recognize videos by `contentType`.

### Example

In case a test involves multiple pages, we could associate a meaningful name with each page, making report analysis easier.

### Motivation

In case of tests involving multiple pages/videos it will make analysis of the report easier.

**Comment by pavelfeldman** [comment]
Makes sense, content-type detection would work better here. Would you like to contribute a pr?

**Comment by RemigiuszDudek** [comment]
Yes, [it is already there](https://github.com/microsoft/playwright/pull/33074) waiting to be merged 😄
